# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasession/MediaSessionController.cpp
# Commit: 5a1aa3d5406e
# Full Hash: 5a1aa3d5406efdb0c8336b152918cc0206366a10
# Author: alwu <alwu@mozilla.com>
# Date: 2020-04-24 03:06:24
# Description:
#   Bug 1631770 - check if `info` exists before update the declared playback state. r=bryce
#   
#   One possible place where `SetDeclaredPlaybackState()` would be called without having a info is that, media session changes its declared playback state after CC unlink process that would result in removing correspond `MediaSessionInfo` from the `mMediaSessionInfoMap`.
#   
#   Therefore, replace the assertion with an early return check.
# ==============================================================================

diff -r fbce36fb761f -r 5a1aa3d5406e dom/media/mediasession/MediaSessionController.cpp
--- a/dom/media/mediasession/MediaSessionController.cpp	Thu Apr 23 19:29:48 2020 +0000
+++ b/dom/media/mediasession/MediaSessionController.cpp	Thu Apr 23 02:30:22 2020 +0000
@@ -209,8 +209,9 @@
 
 void MediaSessionController::SetDeclaredPlaybackState(
     uint64_t aSessionContextId, MediaSessionPlaybackState aState) {
-  MOZ_ASSERT(mMediaSessionInfoMap.Contains(aSessionContextId),
-             "Update declared state for unknown media session!");
+  if (!mMediaSessionInfoMap.Contains(aSessionContextId)) {
+    return;
+  }
   MediaSessionInfo* info = mMediaSessionInfoMap.GetValue(aSessionContextId);
   LOG("SetDeclaredPlaybackState from %s to %s",
       ToMediaSessionPlaybackStateStr(info->mDeclaredPlaybackState),
